gapic/src/main/com/google/gapid/models/Devices.java

```diff
@@ -446,7 +446,7 @@ public DeviceValidationResult(Service.Error error, boolean passed, boolean skipp
     }
 
     public DeviceValidationResult(Service.ValidateDeviceResponse r) {
-      this(r.getError(), !r.hasError(), false);
+      this(r.getError(), !r.hasError() && r.getResult().getValidationErrorMsg().length() == 0, false);
```

Comment:
Just a note for the future: when you update the UI to show more detail, I think we should change `Client.java` to not leak the `*Response` message and turn the error into an exception like we do for other requests. But for this PR, this is OK.